lib/destination/dml/merge.go

```diff
@@ -127,7 +99,7 @@ func (m *MergeArgument) buildRedshiftStatements() ([]string, error) {
 
 	if m.SoftDelete {
 		return []string{
-			m.buildRedshiftInsertQuery(m.Columns),
+			redshiftDialect.RedshiftDialect{}.BuildMergeInsertQuery(m.TableID, m.SubQuery, m.PrimaryKeys, m.Columns),
```

Comment:
I'm assuming this is whole function (generating the merge statement) will move into the Redshift dialect at some point and this is just the interim state?